src/bpm/jobid/encoding.go

```diff
@@ -0,0 +1,88 @@
+// Copyright (C) 2019-Present CloudFoundry.org Foundation, Inc. All rights reserved.
+//
+// This program and the accompanying materials are made available under
+// the terms of the under the Apache License, Version 2.0 (the "License”);
+// you may not use this file except in compliance with the License.
+//
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+// License for the specific language governing permissions and limitations
+// under the License.
+
+package jobid
+
+import (
+	"encoding/hex"
+	"fmt"
+	"strings"
+)
+
+const (
+	alphanumeric = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
+	symbols      = "_-" // . is not included because we use it for escaping
+	combined     = alphanumeric + symbols
+)
+
+// Encode encodes an arbitrary container name into a format which is a valid
+// container name for runc. The encoding must not be depended upon and can
+// change unexpectedly between versions of BPM.
+func Encode(name string) string {
+	id := []byte{}
```

Comment:
This can probably be just a string?

Reply:
Yeah makes sense. The append operation is casting what's already a string to a byte slice, so it makes more sense to just operate on strings.